src/main/kotlin/io/easybreezy/project/application/project/command/Validation.kt

```diff
@@ -53,4 +61,49 @@ class Validation @Inject constructor(private val repository: Repository) {
             validate(WriteDescription::description).isNotBlank()
         }
     }
+
+    object NoParentCategory : Constraint {
+        override val name: String
+            get() = "No category found"
+    }
+
+    suspend fun validateCommand(command: NewCategory): List<Error> {
+        return transactionManager {
+            validate(command) {
+                validate(NewCategory::name).hasSize(2, 25)
+                validate(NewCategory::parent).isNullOrProjectCategory(command.project)
+            }
+        }
+    }
+
+    suspend fun validateCommand(command: ChangeCategory): List<Error> {
+        return transactionManager {
+            validate(command) {
+                validate(ChangeCategory::name).hasSize(0, 25)
```

Comment:
What does 0 mean here? That category name can be blank `""`?

Reply:
Changed to 2,25